File: tests/integration/benchmarks/fixtures/go/rosetta/src/processor/processor_test.go
// Package processor tests for the processor module.
//
// This test file exercises name-matched test detection:
// processor_test.go should automatically back processor.go.
package processor

import (
	"rosetta/src/models"
	"testing"
)

func TestRun(t *testing.T) {
	t.Run("processes records and returns report", func(t *testing.T) {
		records := []models.Record{
			models.CreateRecord(1, "A", 100),
			models.CreateRecord(2, "B", 200),
			models.CreateRecord(3, "C", 150),
		}

		report := Run(records, nil)

		if report.Total != 450 {
			t.Errorf("expected total 450, got %f", report.Total)
		}
		if report.Average != 150 {
			t.Errorf("expected average 150, got %f", report.Average)
		}
		if len(report.Records) != 3 {
			t.Errorf("expected 3 records, got %d", len(report.Records))
		}
	})

	t.Run("handles empty record set", func(t *testing.T) {
		report := Run([]models.Record{}, nil)

		if report.Total != 0 {
			t.Errorf("expected total 0, got %f", report.Total)
		}
		if len(report.Records) != 0 {
			t.Errorf("expected 0 records, got %d", len(report.Records))
		}
	})
}

func TestSummarize(t *testing.T) {
	t.Run("formats report as human-readable string", func(t *testing.T) {
		report := models.Report{
			Total:   450,
			Average: 150,
			Records: []models.Record{},
		}

		summary := Summarize(&report)

		if len(summary) == 0 {
			t.Error("expected non-empty summary")
		}
	})
}

File: tests/integration/benchmarks/fixtures/java/rosetta/src/com/rosetta/processor/ProcessorTest.java
package com.rosetta.processor;

import com.rosetta.models.Record;
import com.rosetta.models.Report;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Unit tests for the Processor class.
 *
 * This test file exercises name-matched test detection:
 * ProcessorTest.java should automatically back Processor.java.
 */
public class ProcessorTest {

    @Nested
    @DisplayName("run()")
    class RunTests {

        @Test
        @DisplayName("processes records and returns report")
        void processesRecordsAndReturnsReport() {
            List<Record> records = Arrays.asList(
                Record.create(1, "A", 100.0),
                Record.create(2, "B", 200.0),
                Record.create(3, "C", 150.0)
            );

            Report report = Processor.run(records, null);

            assertEquals(450.0, report.getTotal(), 0.001);
            assertEquals(150.0, report.getAverage(), 0.001);
            assertEquals(3, report.getRecords().size());
        }

        @Test
        @DisplayName("handles empty record set")
        void handlesEmptyRecordSet() {
            Report report = Processor.run(Collections.emptyList(), null);

            assertEquals(0.0, report.getTotal(), 0.001);
            assertEquals(0, report.getRecords().size());
        }
    }

    @Nested
    @DisplayName("summarize()")
    class SummarizeTests {

        @Test
        @DisplayName("formats report as human-readable string")
        void formatsReportAsHumanReadableString() {
            Report report = new Report(450.0, 150.0, Collections.emptyList());

            String summary = Processor.summarize(report);

            assertNotNull(summary);
            assertFalse(summary.isEmpty());
        }
    }
}
